### PR TITLE
Updated forge-api-nodejs-client to 0.9.x

### PR DESCRIPTION
**Pull request by reediculous456**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/Autodesk-Forge/forge-api-nodejs-client/compare/v0.8.4...v0.9.5
- [X] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## forge-apis ([<kbd>unpkg</kbd>](https://unpkg.com/browse/forge-apis@latest/))
was missing the following properties:
1. ApiClient
1. BadInput
1. BaseAttributesCreatedUpdated
1. BaseAttributesCreatedUpdatedAttributes
1. BaseAttributesExtensionObject

<details>
<summary>as well as these 140 other properties...</summary>
<p>BaseAttributesExtensionObjectWithoutSchemaLink, Bucket, BucketObjects, Buckets, BucketsItems, Conflict, CreateItem, CreateItemData, CreateItemDataRelationships, CreateItemDataRelationshipsTip, CreateItemDataRelationshipsTipData, CreateItemIncluded, CreateItemRelationships, CreateItemRelationshipsStorage, CreateItemRelationshipsStorageData, CreateRef, CreateRefData, CreateRefDataMeta, CreateStorage, CreateStorageData, CreateStorageDataAttributes, CreateStorageDataRelationships, CreateStorageDataRelationshipsTarget, CreateVersion, CreateVersionData, CreateVersionDataRelationships, CreateVersionDataRelationshipsItem, CreateVersionDataRelationshipsItemData, Diagnostics, Folder, FolderAttributes, FolderRelationships, Forbidden, Formats, FormatsFormats, Hub, HubAttributes, HubRelationships, Hubs, Item, ItemAttributes, ItemCreated, ItemRelationships, Job, JobAcceptedJobs, JobDwgOutputPayload, JobDwgOutputPayloadAdvanced, JobIfcOutputPayload, JobIfcOutputPayloadAdvanced, JobIgesOutputPayload, JobIgesOutputPayloadAdvanced, JobObjOutputPayload, JobObjOutputPayloadAdvanced, JobPayload, JobPayloadInput, JobPayloadItem, JobPayloadOutput, JobPayloadMisc, JobPayloadDestination, JobStepOutputPayload, JobStepOutputPayloadAdvanced, JobStlOutputPayload, JobStlOutputPayloadAdvanced, JobSvfOutputPayload, JobSvfOutputPayloadAdvanced, JobSvf2OutputPayload, JobSvf2OutputPayloadAdvanced, JobThumbnailOutputPayload, JobThumbnailOutputPayloadAdvanced, JsonApiAttributes, JsonApiCollection, JsonApiDocument, JsonApiDocumentBase, JsonApiError, JsonApiErrorErrors, JsonApiErrorLinks, JsonApiLink, JsonApiLinks, JsonApiLinksPaging, JsonApiLinksRelated, JsonApiLinksSelf, JsonApiMeta, JsonApiMetaLink, JsonApiRelationships, JsonApiRelationshipsLinksExternalResource, JsonApiRelationshipsLinksInternal, JsonApiRelationshipsLinksInternalResource, JsonApiRelationshipsLinksRefs, JsonApiRelationshipsLinksRefsLinks, JsonApiResource, JsonApiTypeId, JsonApiVersion, JsonApiVersionJsonapi, Manifest, ManifestChildren, ManifestDerivative, Message, Messages, Metadata, MetadataCollection, MetadataData, MetadataMetadata, MetadataObject, NotFound, ObjectDetails, ObjectFullDetails, ObjectFullDetailsDeltas, ObjectS3Download, ObjectS3Upload, ObjectCompleteS3Upload, Permission, PostBucketsPayload, PostBucketsPayloadAllow, PostBucketsSigned, PostObjectSigned, Project, ProjectAttributes, ProjectRelationships, Projects, Reason, Refs, RelRef, RelRefMeta, ResourceId, Result, Storage, StorageCreated, StorageRelationships, StorageRelationshipsTarget, StorageRelationshipsTargetData, TopFolders, UserProfile, UserProfileProfileImages, Version, VersionAttributes, VersionCreated, VersionRelationships, Versions, IssuesApi, setDebug</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against f8fddce0dc217df6fe3519bcc0bbc8e129e109d2
</p>

File: types/forge-apis/index.d.ts [hunk 1]
```diff
@@ -60,7 +60,7 @@ export interface Credentials {
     client_id: string;
     client_secret: string;
     grant_type: string;
-    scope?: Scope | undefined;
```
Comment (jakebailey) [inline types/forge-apis/index.d.ts:63]: Are you removing these to fix a bug? If not, these shouldn't go away, as they exist as to support `exactOptionalPropertyTypes`.

See: [Playground Link](https://www.typescriptlang.org/play?exactOptionalPropertyTypes=true&ts=4.4.2#code/PTAEHUEsBcAtQKYA8CGBjaB5ADtSB7AOxQBsAFAJ32wQugE8AVemgZ0WICMSEATAGlDMaAZTQVIuALAAoEKFgp2raBIy1QFAK492KKlsK9QAd0XRQ9fFtBoUhUKwQX7oaniKlZ87FRp16UGh8ADpZbzAAMXwKRFQAW2weQThIdkhCaFoAM3QEUF4ENBJ9BHY4FAs4Wny00BQI0F9qWgZTWEg0eDsHTnyifuygk3xHVQyAc1YALlAAcl59AGs5t1i5kkgJ2Gg5xpjQGEdYaxJjQnwLPsOHardOACsi6DCZWQysily0fIBVJwoABEELkdNB2ABvWSgUDyRiwfIoThOQg-NxDFCgABupC0+QoCF8ZQQmXYc1Y9BUCHiexkMLQ+BIMXh1IQmCxtAkhQA-LMAESLChLPmgAA+oD5m220D5AG5ZABfcJyMBQODWKoddjZEoTDhIngCIIIgn1U1wAn5HEkPHsMydeBWGw9RpOCwAAwZTIoLPibI5FC5CHdQVGfX5guFgklWx2IvsxndhkK2QyfHdr0aImceEIeru1rxodASaMILTvBDJkgJBI9Vr+BMoHi+BUoAAUigcWIJLhNIY8H7Gl0ikttQc7sg0llUflgqBcjXBPauod2CDss9IBySIFciQKa95AAJRsIAMpE0Icn1BdaABe97TrHKo1S7D6iixBAoISECIKctCBgAhCEabIDk9RlmQRP12U5SBChDAB9Xh8DKQg5mgZCbmKLRChLZMgPTTMVX-fIdRQPV3WQdAsFwUDSEoFoAmEMoQ3iFAljKIQWAQHtJBcNA0C0ChKgQXdGk46BVzuFMMhAogmiob9CmMJR6gcdxGLrZp-AYUjCmKUo71RDwHAmZx-lobNoFzKYAAoAEpZmsoEQRQMFWHlN4ZAZQg2zdez2AAXlASzoDc2zgucnygsmVgQi9GDWXgwNEPyMKBWWOVZHivNEuSn1YP9BCCKyqU4x8xoACEtAsC5oGmPKcwSpLoOK1KAyDUAwqI1NCD4aqyMoH9izuSjqNojAcHMpi-FaJg+NYd1LzqEwNMappW1YSBuAQEIgA)

**Review by jakebailey — CHANGES_REQUESTED** [review]
I don't think you should remove `| undefined`; if anything, the new types should say `| undefined` more.

**Review by jakebailey — APPROVED** [review]

**Comment by reediculous456** [comment]
Ready to merge

**Pull request merged**